packages/sources/oanda/src/endpoint/price.ts

```diff
@@ -0,0 +1,153 @@
+import {
+  EndpointContext,
+  PriceEndpoint,
+  priceEndpointInputParameters,
+  PriceEndpointParams,
+} from '@chainlink/external-adapter-framework/adapter'
+import { RoutingTransport } from '@chainlink/external-adapter-framework/transports/meta'
+import { HttpTransport } from '@chainlink/external-adapter-framework/transports'
+import { makeLogger } from '@chainlink/external-adapter-framework/util'
+import { AdapterError } from '@chainlink/external-adapter-framework/validation/error'
+
+import Decimal from 'decimal.js'
+import axios from 'axios'
+
+import restPairs from '../config/restPairs.json'
+import { ModifiedSseTransport } from '../transport/modifiedSSE'
+import {
+  EndpointTypes,
+  HttpGenerics,
+  InstrumentList,
+  InstrumentMap,
+  ModifiedSseGenerics,
+  RestPairs,
+} from '../types'
+
+const logger = makeLogger('OandaPrice')
+
+let instrumentMap: InstrumentMap
+
+// Get mapping of all available instruments keyed by base and quote assets
+const setInstrumentMap = async (context: EndpointContext<ModifiedSseGenerics>) => {
+  logger.info({ msg: 'Setting instrument map' })
+  const { data, status } = await axios.get<InstrumentList>(
+    `${context.adapterConfig.INSTRUMENTS_API_ENDPOINT}/accounts/${context.adapterConfig.API_ACCOUNT_ID}/instruments`,
+    {
+      headers: {
+        contentType: 'application/json',
+        Authorization: `Bearer ${context.adapterConfig.SSE_API_KEY}`,
+      },
+    },
+  )
+
+  if (!data || status !== 200) {
+    throw new AdapterError({ message: 'Could not fetch asset list', providerStatusCode: status })
```

Comment:
We could use `AdapterDataProviderError` here since the DP is returning the error. I believe it'll have you provide timestamps but for `providerIndicatedTimeUnixMs` you can mark as undefined since it wouldn't make sense here.